data/modifications/pick_advance/main.lua

```diff
@@ -220,6 +219,7 @@ on_event("moveto", function()
 				initialize_unit(unit)
 			end
 		end
+		wesnoth.allow_undo(false)
```

Comment:
> check if there are any new units that need to be forced to make an advancement choice

So if there are no units then no need for allow_undo()?

Reply:
it still changes the `pa_fresh_turn` variable,